### PR TITLE
add e2e tests

### PR DESCRIPTION
**Pull request by gykuma**
Added e2e tests for build artifacts

### CODE REVIEW

File: Extensions/ArtifactEngine/E2ETests/E2E.ts [hunk 1]
```diff
@@ -47,11 +50,76 @@ describe('e2e tests', () => {
...
                 throw "test failure";
             });
     });
+
+    it('should be able to download build artifact', function (done) {
+        this.timeout(15000);
+        let processor = new engine.ArtifactEngine();
+
+        let processorOptions = new engine.ArtifactEngineOptions();
+        processorOptions.fileProcessingTimeoutInMinutes = 5;
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/E2ETests/E2E.ts]: fileProcessingTimeoutInMinutes is not used by artifactEngine. can you confirm with Omesh & remove it if this is the case?

File: Extensions/ArtifactEngine/E2ETests/E2E.ts [hunk 2]
```diff
@@ -47,11 +50,76 @@ describe('e2e tests', () => {
...
+
+                assert.equal(tickets.find(x => x.artifactItem.path == "dropz/file1.pdb").retryCount, 0);
+                assert.equal(tickets.find(x => x.artifactItem.path == "dropz/folder1/file2.txt").retryCount, 0);
+                done();
+            }, (error) => {
+                throw "test failure";
+            });
+    });
+
+    it('should be able to download large build artifact', function (done) {
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/E2ETests/E2E.ts]: good one! what is the size of the build artifact?

**Review by GitHubSriramB — APPROVED** [review]

File: Extensions/ArtifactEngine/E2ETests/E2E.ts (hunk 2 above)
Comment (gykuma) [reply Extensions/ArtifactEngine/E2ETests/E2E.ts]: its around 1.7 GBs with 301 files

File: Extensions/ArtifactEngine/E2ETests/E2E.ts (hunk 1 above)
Comment (gykuma) [reply Extensions/ArtifactEngine/E2ETests/E2E.ts]: removed

File: Extensions/ArtifactEngine/E2ETests/E2E.ts [hunk 3]
```diff
@@ -47,11 +49,48 @@ describe('e2e tests', () => {
                     assert.equal(data, "dummyFolderContent");
                     done();
                 });
-                
+
                 assert.equal(tickets.find(x => x.artifactItem.path == "Extensions/ArtifactEngine/TestData/Jenkins/file1.pdb").retryCount, 0);
                 assert.equal(tickets.find(x => x.artifactItem.path == "Extensions/ArtifactEngine/TestData/Jenkins/folder1/file2.txt").retryCount, 0);
             }, (error) => {
                 throw "test failure";
```
Comment (omeshp) [inline Extensions/ArtifactEngine/E2ETests/E2E.ts]: throw error

File: Extensions/ArtifactEngine/E2ETests/E2E.ts [hunk 4]
```diff
@@ -47,11 +49,48 @@ describe('e2e tests', () => {
...
+                    }
+                    assert.equal(data, "dummyFolderContent");
+                    done();
+                });
+
+                assert.equal(tickets.find(x => x.artifactItem.path == "dropz/file1.pdb").retryCount, 0);
+                assert.equal(tickets.find(x => x.artifactItem.path == "dropz/folder1/file2.txt").retryCount, 0);
+                done();
+            }, (error) => {
+                throw "test failure";
```
Comment (omeshp) [inline Extensions/ArtifactEngine/E2ETests/E2E.ts]: throw error

File: Extensions/ArtifactEngine/PerfTests/Perf.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+        var sourceProvider = new providers.FilesystemProvider(itemsUrl);
+        var dropLocation = path.join(config.dropLocation, "fileshareWithLargeFiles");
+        var filesystemProvider = new providers.FilesystemProvider(dropLocation);
+
+        processor.processItems(sourceProvider, filesystemProvider, processorOptions)
+            .then((tickets) => {
+                assert.equal(tickets.filter(x => x.artifactItem.itemType === ItemType.File && x.state === TicketState.Processed).length, 301);
+                done();
+            });
```
Comment (omeshp) [inline Extensions/ArtifactEngine/PerfTests/Perf.ts]: add error block.

File: Extensions/ArtifactEngine/PerfTests/Perf.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        var itemsUrl = "https://testking123.visualstudio.com/_apis/resources/Containers/1902716?itemPath=largedrop&isShallow=false";
+        var variables = {};
+
+        var handler = new PersonalAccessTokenCredentialHandler(config.vsts.pat);
+        var webProvider = new providers.WebProvider(itemsUrl, "vsts.handlebars", variables, handler, { ignoreSslError: false });
+        var dropLocation = path.join(config.dropLocation, "vstsDropWithLargeFiles");
+        var filesystemProvider = new providers.FilesystemProvider(dropLocation);
+
+        processor.processItems(webProvider, filesystemProvider, processorOptions)
+            .then((tickets) => {
```
Comment (omeshp) [inline Extensions/ArtifactEngine/PerfTests/Perf.ts]: add error block.

File: Extensions/ArtifactEngine/PerfTests/Perf.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+describe('perf tests', () => {
+    it('should be able to download large build artifact from vsts drop', function (done) {
+        this.timeout(300000);
+        let processor = new engine.ArtifactEngine();
+
+        let processorOptions = new engine.ArtifactEngineOptions();
+        processorOptions.itemPattern = "**";
+        processorOptions.parallelProcessingLimit = 8;
+        processorOptions.retryIntervalInSeconds = 1000;
+        processorOptions.retryLimit = 5;
```
Comment (omeshp) [inline Extensions/ArtifactEngine/PerfTests/Perf.ts]: retry limit 4

File: Extensions/ArtifactEngine/PerfTests/Perf.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+    it('should be able to download large build artifact from fileshare', function (done) {
+        this.timeout(900000);
+        let processor = new engine.ArtifactEngine();
+
+        let processorOptions = new engine.ArtifactEngineOptions();
+        processorOptions.itemPattern = "**";
+        processorOptions.parallelProcessingLimit = 8;
+        processorOptions.retryIntervalInSeconds = 1000;
+        processorOptions.retryLimit = 5;
```
Comment (omeshp) [inline Extensions/ArtifactEngine/PerfTests/Perf.ts]: retry limit 4

File: Extensions/ArtifactEngine/E2ETests/E2E.ts [hunk 9]
```diff
@@ -17,14 +20,13 @@ describe('e2e tests', () => {
         let processor = new engine.ArtifactEngine();
 
         let processorOptions = new engine.ArtifactEngineOptions();
-        processorOptions.fileProcessingTimeoutInMinutes = 5;
         processorOptions.itemPattern = "**";
         processorOptions.parallelProcessingLimit = 8;
         processorOptions.retryIntervalInSeconds = 1000;
```
Comment (omeshp) [inline Extensions/ArtifactEngine/E2ETests/E2E.ts]: retryIntervalInSeconds = 2 secs

**Review by omeshp — APPROVED** [review]

File: Extensions/ArtifactEngine/PerfTests/Perf.ts (hunk 6 above)
Comment (gykuma) [reply Extensions/ArtifactEngine/PerfTests/Perf.ts]: added

File: Extensions/ArtifactEngine/PerfTests/Perf.ts (hunk 5 above)
Comment (gykuma) [reply Extensions/ArtifactEngine/PerfTests/Perf.ts]: added

File: Extensions/ArtifactEngine/E2ETests/E2E.ts [hunk 10]
```diff
@@ -47,11 +49,74 @@ describe('e2e tests', () => {
...
+
+                assert.equal(tickets.find(x => x.artifactItem.path == "dropz/file1.pdb").retryCount, 0);
+                assert.equal(tickets.find(x => x.artifactItem.path == "dropz/folder1/file2.txt").retryCount, 0);
+                done();
+            }, (error) => {
+                throw "test failure";
+            });
+    });
+
+    it('should be able to download large build artifact', function (done) {
```
[the remark this answers is not in the document]
Comment (gykuma) [reply Extensions/ArtifactEngine/E2ETests/E2E.ts]: added within perfTests

[the remark this answers is not in the document]
Comment (gykuma) [reply Extensions/ArtifactEngine/E2ETests/E2E.ts]: moved to perfTests

**Pull request merged**